luaui/Widgets/gui_top_bar.lua

```diff
@@ -644,6 +649,73 @@ local function updateWind()
 	end
 end
 
+-- return true if tidal speed is *relevant*, enough water in the world (>= 10%)
+local function checkTidalRelevant()
+   water = 0
+   nowater = 0
+   for x = 0, Game.mapSizeX do
```

Comment:
What is the size of each step? How much area does this sweep perform against the whole area of the map.

If too granular you perform unnecessary calculation, if too coarse you get unreliable information

Reply:
this can be done another way...

local _, _, mapMinHeight, mapMaxHeight = Spring.GetGroundExtremes()
	if mapMinHeight <= -2 then

Reply:
??

Reply:
GetGroundExtremes can mean there is a 8x8 elmo puddle of water on a 8192x8192 elmo map though, doesnt necessarily mean viable tidals

Reply:
Right now it goes through *each* coordinate, but it could be changed to just go through every 10 or even 100 ... 
`for x = 0, Game.mapSizeX, 10`